lib/skate/detours/detour.ex

```diff
@@ -30,6 +30,59 @@ defmodule Skate.Detours.Detour do
       :author_id,
       :status
     ]
+
+    def from(
+          status,
+          %{
+            state: %{
+              "context" => %{
+                "route" => %{"name" => route_name, "directionNames" => direction_names},
+                "routePattern" => %{"headsign" => headsign, "directionId" => direction_id},
+                "nearestIntersection" => nearest_intersection
+              }
+            }
+          } = db_detour
+        ) do
```

Comment:
I like moving this logic here! I'm trying to think of an alternative name but don't have a great one... Maybe an explanatory comment would be good though? I see `from` and assumed it was something like a kernel operation I didn't know about

Reply:
I personally would like to keep this convention (and do it more 😅), I _did_ pick it up from Rust [which got inspiration from some other language IIRC] (there's a From/Into convention, where you can do 
```rs
1.into<String>()
// or 
String::from(1)
```
and I think this is a good way to do conversions.

Reply:
It could be named `from_detour_snapshot` though. I think `from` is still the best descriptor/word though.

Reply:
just a note: I'd recommend leaning on Elixir LS and VSCode LSP tooling for that kind of search, because you can do a "Find all references" search which uses the information from the compiler to find _actual_ references based on the compiled code, rather than text search.